src/extension.js

```diff
@@ -55,25 +58,216 @@ module.exports.activate = function( context ) {
 					// The task's source, in our case our registered name of 'testbox'
 					"testbox",
 					// Shell execution via the command output() method
-					new vscode.ShellExecution( globalCommand.output ),
+					new vscode.ShellExecution(globalCommand.output),
 					// The problem matcher id => package.json
 					"$testbox"
 				)
 			];
 		}
-	} ) );
+	}));
+
+	// Add the UI Panel for the TestBox Runner
+	const controller = vscode.tests.createTestController('cfmlTestController', 'CFML Tests');
+
+	disposables.push(controller);
+	discoverTests(controller);
+	// What do we match for tests?
+	// TODO: externalize the glob pattern
+	const watcher = vscode.workspace.createFileSystemWatcher(testFileGlob);
+	watcher.onDidCreate(() => discoverTests(controller));
+	watcher.onDidChange(() => discoverTests(controller));
+	watcher.onDidDelete(() => discoverTests(controller));
+
+
+
+	// Also change on config
+	disposables.push(
+		vscode.workspace.onDidChangeConfiguration(e => {
+			// We could filter this out more but for now just re-discover the tests
+			// if (
+			// 	e.affectsConfiguration('testbox.excludedPaths') ||
+			// 	e.affectsConfiguration('testbox.pathMappings') ||
+			// 	e.affectsConfiguration('testbox.runnerUrl')
+			// ) {
+			discoverTests(controller);
+			// }
+		})
+	);
+
+
+	disposables.push(watcher);
+
+
+	// Create a run profile that runs tests
+	controller.createRunProfile('Run Tests', vscode.TestRunProfileKind.Run, (request, token) => {
+		runTestsViaURL(request, token, controller);
+	});
+
+
+	// Listen for configuration changes
+
+	context.subscriptions.push(disposables);
+	LOG.info("Congratulations, the \"TestBox\" Extension is now active!");
 
-	// Add all commands
-	context.subscriptions.push( disposables );
 };
 
+// Run raw, without box command abstraction
+/**
+ * Runs tests via a specified URL rather than using the BoxCommand abstraction as it willl be faster and the user doesnt need to 
+ * install commandbox to run tests
+ *
+ * @param {vscode.TestRunRequest} request - The test run request.
+ * @param {vscode.CancellationToken} token - The cancellation token.
+ * @param {vscode.TestController} controller - The test controller.
+ * @returns {Promise<void>} A promise that resolves when the tests have been run.
+ */
+async function runTestsViaURL(request, token, controller) {
```

Comment:
I use the runner directly since using it through box + testbox is rather slow. This is fine in the commandline etc, but much faster if I am running a bunch of tests.